sdks/go/pkg/beam/core/typex/class.go

```diff
@@ -91,16 +92,25 @@ func ClassOf(t reflect.Type) Class {
 // data must be fully serializable. Functions and channels are examples of invalid
 // types. Aggregate types with no universals are considered concrete here.
 func IsConcrete(t reflect.Type) bool {
+	concrete, _ := isConcrete(t, make(map[uintptr]bool))
+	return concrete
+}
+
+// AssertConcrete returns true iff the given type is a valid "concrete" data type and if not,
+// returns false along with an error indicating why the type is not concrete. Concrete
+// data must be fully serializable. Functions and channels are examples of invalid
+// types. Aggregate types with no universals are considered concrete here.
+func AssertConcrete(t reflect.Type) (bool, error) {
```

Comment:
Assert generally implies that something will panic or exit, rather than return an error. Perhaps `CheckConcrete` instead?

Reply:
CheckConcrete works for me!